### DISCUSSION TITLE
Trying to extend the inventserialid length on the POS

### DISCUSSION THREAD

**Discussion by StefB1995** [microsoft/Dynamics365Commerce.Solutions#337] [Q&A]
Hi all,

At our customer we need to increase the size of the serial number that can be entered.

In the documentation ([link](https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/extended-columns)) I found the following:

The length of the INVENTSERIALID field in the channel database is 50 characters, but the POS UI accepts serial numbers that are a maximum of only 20 characters long. In this case, you must extend the field in both the POS UI and the Commerce Headquarters database.

Does anyone have experience on how to extend the field in the POS UI, I don't find much information in the documentation on how to do this?

Already created a ticket at Microsoft for more information but they're still looking at it.

 
Thanks in advance!

**Comment by lukedgr** [comment]
Hi @StefB1995,

It is not supported to extend the field in the serial number dialog itself, however you can replace the entire dialog used to capture the serial number by [replacing the handler for the GetSerialNumberClientRequest](https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/pos-handler).

When you add an item to a cart, a dialog standard is asking to fill Serial Number, if the item needs to, the code below, overwrite that and I am forcing and hardcoded the Serial Number, they can do a custom dialog to user provide the serial number, instead of hardcoded it. 

If customer needs a sample code of a Custom dialog, in GasStation sample we have it.

[Dynamics365Commerce.InStore/src/StoreCommerceSamples/Solutions/GasStation/Pos/Create/Dialogs at release/9.51 · microsoft/Dynamics365Commerce.InStore · GitHub](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FDynamics365Commerce.InStore%2Ftree%2Frelease%2F9.51%2Fsrc%2FStoreCommerceSamples%2FSolutions%2FGasStation%2FPos%2FCreate%2FDialogs&data=05%7C02%7CLuke.Graham%40microsoft.com%7C8f1f212cf2194597973c08dd08a463cc%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638676224484147757%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=%2Bku49%2BlxFAWSAxLIjN3baocbbChVhQm%2Bcopv8SaZZlo%3D&reserved=0)

TS sample code from POS/SCA:
![image](https://github.com/user-attachments/assets/3b307ae3-3cfd-4a3d-a118-a1977d8fdd17)

With the custom dialog you can remove any length validation for the serial number to work through the 20 character limit.

An example on how to create a custom dialog can be found [here](https://github.com/microsoft/Dynamics365Commerce.InStore/blob/d79248181301bd1b16a4494a902ccc9624e46713/src/StoreCommerceSamples/PosExtensionSamples/Pos/Create/Dialogs/BarcodeMsrDialog/BarcodeMsrDialog.ts#L10)
